File: python/1003.py
from sys import stdin
def fzro(n):
    if n==0:
        return 1
    elif n==1:
        return 0
    elif zro[n]!=0:
        return zro[n]
    zro[n] = fzro(n-1) + fzro(n-2)
    return zro[n]

def fone(n):
    if n==0:
        return 0
    elif n==1:
        return 1
    elif one[n]!=0:
        return one[n]
    one[n] = fone(n-1) + fone(n-2)
    return one[n]

zro = list()
one = list()
t = int(stdin.readline())
for i in range(t):
    n = int(stdin.readline())
    zro = [0 for i in range(n + 2)]
    one = [0 for i in range(n + 2)]
    print('%d %d' %(fzro(n), fone(n)))
    

File: baekjoon/c/1003.c
#define _CRT_SECURE_NO_WARNINGS

#include <stdio.h>
#include <stdlib.h>

int zro[41] = { 0 }, one[41] = { 0 };
int fzro(int);
int fone(int);

int main()
{
	int t;
	scanf("%d", &t);
	while (t--)
	{
		int n;
		scanf("%d", &n);
		printf("%d %d\n", fzro(n), fone(n));
	}
}

int fzro(int n)
{
	if (n == 0)
		return 1;
	else if (n == 1)
		return 0;
	else if (zro[n] != 0)
			return zro[n];
	return zro[n] = fzro(n - 1) + fzro(n - 2);
}

int fone(int n)
{
	if (n == 0)
		return 0;
	else if (n == 1)
		return 1;
	else if (one[n] != 0)
		return one[n];
	return one[n] = fone(n - 1) + fone(n - 2);
}